snapcraft/plugins/plainbox_provider.py

```diff
@@ -46,3 +49,8 @@ def build(self):
             "python3", "manage.py", "install", "--layout=relocatable",
             "--prefix=/providers/{}".format(self.name),
             "--root={}".format(self.installdir)])
+
+        # Fix all shebangs to use the in-snap python.
+        file_utils.replace_in_file(self.installdir, re.compile(r''),
+                                   re.compile(r'#!.*python'),
```

Comment:
I'd like to see this be r'^#!.*python' or something similar so that in the rare case that that string is in a file it won't get replaced.  May need to take leading spaces into account as well, IDK what posix says about shebang lines.